src/containers/CreateCustomRun/CreateCustomRun.jsx

```diff
@@ -0,0 +1,484 @@
+
+const YAMLEditor = React.lazy(() => import('../YAMLEditor'));
+
+const initialState = {
+  creating: false,
+  customRef: '',
+  customRunName: '',
+  customSpec: '',
+  invalidLabels: {},
+  kind: 'CustomRun',
+  labels: [],
+  namespace: '',
+  params: {},
+  paramSpecs: [],
+  retries: '',
+  serviceAccountName: '',
+  submitError: '',
+  timeout: '',
+  validationError: false,
+  validCustomRunName: true,
+  workspaces: ''
+};
+
+const initialParamsState = paramSpecs => {
+  if (!paramSpecs) {
+    return {};
+  }
+  return paramSpecs.reduce(
+    (acc, param) => ({ ...acc, [param.name]: param.default || '' }),
+    {}
+  );
+};
+
+const itemToString = ({ text }) => text;
+
+function CreateCustomRun() {
+  const intl = useIntl();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { selectedNamespace: defaultNamespace } = useSelectedNamespace();
+
+  function getCustomDetails() {
+    const urlSearchParams = new URLSearchParams(location.search);
+    return {
+      kind: urlSearchParams.get('kind') || 'Custom',
+      customName: urlSearchParams.get('customName') || ''
+    };
+  }
+
+  function getCustomRunName() {
+    const urlSearchParams = new URLSearchParams(location.search);
+    return urlSearchParams.get('customRunName') || '';
+  }
+
+  function getNamespace() {
+    const urlSearchParams = new URLSearchParams(location.search);
+    return (
+      urlSearchParams.get('namespace') ||
+      (defaultNamespace !== ALL_NAMESPACES ? defaultNamespace : '')
+    );
+  }
+
+  function isYAMLMode() {
+    const urlSearchParams = new URLSearchParams(location.search);
+    return urlSearchParams.get('mode') === 'yaml';
+  }
+
+  const { kind: initialCustomKind, customName: customRefFromDetails } =
+    getCustomDetails();
+  const [
+    {
+      creating,
+      customRef,
+      customRunName,
+      customSpec,
+      invalidLabels,
+      kind,
+      labels,
+      namespace,
+      params,
+      retries,
+      serviceAccountName,
+      submitError,
+      timeout,
+      validationError,
+      validCustomRunName,
+      workspaces
+    },
+    setState
+  ] = useState({
+    ...initialState,
+    kind: initialCustomKind || 'Custom',
+    namespace: getNamespace(),
+    customRef: customRefFromDetails,
+    params: initialParamsState(null)
+  });
+
+  const { data: custom, error: customError } = useTaskByKind(
+    { kind, name: customRef, namespace },
+    { enabled: !!customRef }
+  );
+
+  const paramSpecs = custom?.spec?.params;
+
+  useTitleSync({
+    page: intl.formatMessage({
+      id: 'dashboard.createCustomRun.title',
+      defaultMessage: 'Create CustomRun'
+    })
+  });
+
+  function switchToYamlMode() {
+    const queryParams = new URLSearchParams(location.search);
+    queryParams.set('mode', 'yaml');
+    const browserURL = location.pathname.concat(`?${queryParams.toString()}`);
+    navigate(browserURL);
+  }
+
+  function checkFormValidation() {
```

Comment:
There's still a lot of code in this component related to the form-based UI, for example, `checkFormValidation`, `handleAddLabel`, `handleNamespaceChange`, `handleSubmit`, etc.

These should all be removed. Then you can also remove any of the unused state, for example `initialParams`, `initialParamsState`, etc.